src/basic/locale-util.c

```diff
@@ -234,24 +234,33 @@ int get_locales(char ***ret) {
         return 0;
 }
 
-bool locale_is_valid(const char *name) {
+int locale_is_valid(const char *name) {
+        _cleanup_(freelocalep) locale_t loc = 0;
```

Comment:
hmm, so the posix docs usually write "(locale_t) 0" instead of just "0" (for example newlocale() documentation), as if locale_t might not necessarily be a type that implicitly can cast from 0, hence would prefer if we'd do the same here, we do this in all other cases afaics